mypy/messages.py

```diff
@@ -819,6 +819,19 @@ def typeddict_instantiated_with_unexpected_items(self,
         self.fail('Expected items {} but found {}.'.format(
             expected_item_names, actual_item_names), context)
 
+    def typeddict_item_name_must_be_string_literal(self,
+                                                   typ: TypedDictType,
+                                                   context: Context):
+        self.fail('Cannot prove expression is a valid item name. Expected one of {}.'.format(
+            list(typ.items.keys())), context)
+
+    def typeddict_item_name_not_found(self,
+                                      typ: TypedDictType,
+                                      item_name: str,
+                                      context: Context):
+        self.fail('\'{}\' is not a valid item name. Expected one of {}.'.format(
```

Comment:
Similar to above: a more consistent message would be `... is not a valid item name; expected one of [...]` (not dot at the end).

Also consider truncating the list (e.g. `['x', 'y', 'z', 'a', 'b', ...]`) if it's long enough.